spec/sasl_authenticator_spec.rb

```diff
@@ -91,4 +93,24 @@
       }.to raise_error(Kafka::FailedScramAuthentication)
     end
   end
+
+  context "when SASL OAuthBearer has been configured" do
```

Comment:
I think there should be a test case for the missing of extensions here.

Reply:
Strange. I added a test case so that I pass in a TokenProvider instance that doesn't have `extensions` defined. I ran `rspec` and that test failed with a TIMEOUT error during Kafka authentication. 

However, when I use my ruby-kafka fork with a test-client I wrote that authenticates to a local Kafka cluster, I pass in a token provider that doesn't have `extensions` defined, and it works fine.